File: 双指针/offer-57.cpp
#include<vector>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int l=0,r=nums.size()-1;
        while(l<r){
            if ((nums[l] + nums[r])==target)
                return {nums[l],nums[r]};
            if ((nums[l]+nums[r])<target)
                l++;
            else
                r--;
        }
        return {};
    }
};

File: 双指针/offer-57.py
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = 0
        r = len(nums)-1
        while(l<r):
            if (nums[l] + nums[r])==target:
                return [nums[l],nums[r]]
            if (nums[l]+nums[r])<target:
                l+=1
            else:
                r-=1
        return []

